spring-amqp/src/main/java/org/springframework/amqp/support/ServerGeneratesConsumerTagStrategy.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.amqp.support;
+
+/**
+ * The default strategy for consumer tags - causes the server to generate the tag
+ * when processing the {@code basicConsume} operation.
+ *
+ * @author Gary Russell
+ * @since 1.4.5
+ *
+ */
+public class ServerGeneratesConsumerTagStrategy implements ConsumerTagStrategy {
+
+	@Override
+	public String createConsumerTag(String queue) {
+		return "";
```

Comment:
Would you mind explaining the reason of such a simple impl?
Looks like the ternary operator on "null check" with fallback to the empty string will be enough.